### PR TITLE
update locale aliases for glibc 2.24

### PR DESCRIPTION
**Pull request by benjaminp**
I also made makelocalealias.py prefer the glibc defaults.

### CODE REVIEW

File: Tools/i18n/makelocalealias.py
Comment (serhiy-storchaka) [inline Tools/i18n/makelocalealias.py:139]: Why this is changed?

**Review by serhiy-storchaka — CHANGES_REQUESTED** [review]
This change needs an entry in Misc/NEWS. Please open an issue on the tracker for discussion and reference.

**Comment by serhiy-storchaka** [comment]
I suppose some of changes are related to bpo-20087. This was not discussed yet.

File: Tools/i18n/makelocalealias.py
Comment (benjaminp) [reply Tools/i18n/makelocalealias.py:139]: Because some of the X11 locale aliases conflict with the glibc ones. glibc obviously uses the glibc ones. This fixes problems where we would return encodings (like `en_IN.ISO8859-1`) that don't correspond to the C-level one.

**Comment by benjaminp** [comment]
I think this is the correct fix for bpo-29571.

File: Tools/i18n/makelocalealias.py
Comment (serhiy-storchaka) [reply Tools/i18n/makelocalealias.py:139]: This is separate issue (bpo-20087). Let don't mix updating to newer glibc with changing the priority of aliases tables. The latter change needs blessing from the module maintainer.

Comment (benjaminp) [reply Tools/i18n/makelocalealias.py:139]: IMO, this is the only interesting change. I only upgraded the alias list because finding an old version of glibc is slightly annoying and updating it is unlikely to break things.

**Comment by ncoghlan** [comment]
Aye, I never did figure out why `locale.getpreferredencoding(False)` and `locale.getlocale(LC_CTYPE)` were giving different answers - this would definitely explain it.

+1 for fixing the underlying discrepancy in 3.7+.

**Comment by serhiy-storchaka** [comment]
@benjaminp, when you run `Tools/i18n/makelocalealias.py` on unmodified sources, you get also a list of modified aliases:
```
#    updated 'az_az' -> 'az_AZ.ISO8859-9E' to 'az_AZ.UTF-8'
#    updated 'ca_ad' -> 'ca_AD.ISO8859-1' to 'ca_AD.ISO8859-15'
#    updated 'ca_fr' -> 'ca_FR.ISO8859-1' to 'ca_FR.ISO8859-15'
```
It should be added to the long comment before the alias table. Only "updated" lines are needed, not "removed".

Add a comment that glibc aliases now are preferred. This is the main cause of changes.

And please add a test for en_IN.

**Comment by malemburg** [comment]
Please see the ticket for discussion: http://bugs.python.org/issue20087

**Review by serhiy-storchaka — APPROVED** [review]

**Comment by malemburg** [comment]
This merge should be undone. The PR clearly contains bugs and also is not complete. See the ticket for discussion.

**Pull request merged**
